tools/data_usage/console.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include <windows.h>
+#include <iostream>
+#include <string>
+#include "ebpf_api.h"
+#include "ebpf_nethooks.h"
+
+#define MAP_COUNT 1
+
+int main(int argc, const char** argv)
+{
+    UNREFERENCED_PARAMETER(argc);
+    UNREFERENCED_PARAMETER(argv);
+
+    ebpf_handle_t flow_program_handle;
+    ebpf_handle_t flow_map_handle[MAP_COUNT];
+    ebpf_handle_t flow_link_handle;
+
+    ebpf_handle_t mac_program_handle;
+    ebpf_handle_t mac_map_handle[MAP_COUNT];
+    ebpf_handle_t mac_link_handle;
+
+    uint32_t count_of_map_handle = MAP_COUNT;
+    const char* error_message = nullptr;
+
+    uint32_t result = ebpf_api_initiate();
+    if (result != ERROR_SUCCESS)
+    {
+        std::cerr << "ebpf_api_initiate failed: " << result << std::endl;
+        return 1;
+    }
+
+    result = ebpf_api_load_program(
+        "associatetoflow.o", "flow", EBPF_EXECUTION_JIT, &flow_program_handle, &count_of_map_handle, flow_map_handle, &error_message);
+
+    if (result != ERROR_SUCCESS)
+    {
+        std::cerr << "Failed to load associatetoflow eBPF program: " << error_message << std::endl;
+        ebpf_free_string(error_message);
+        return 1;
+    }
+
+    result = ebpf_api_link_program(flow_program_handle, EBPF_ATTACH_TYPE_FLOW, &flow_link_handle);
+    if (result != ERROR_SUCCESS)
+    {
+        std::cerr << "Failed to attach associatetoflow eBPF program: " << result << std::endl;
+        return 1;
+    }
+
+    result = ebpf_api_load_program(
+        "countbytes.o", "mac", EBPF_EXECUTION_JIT, &mac_program_handle, &count_of_map_handle, mac_map_handle, &error_message);
+
+    if (result != ERROR_SUCCESS)
+    {
+        std::cerr << "Failed to load countbytes eBPF program: " << error_message << std::endl;
+        ebpf_free_string(error_message);
+        return 1;
+    }
+
+    result = ebpf_api_link_program(mac_program_handle, EBPF_ATTACH_TYPE_MAC, &mac_link_handle);
+    if (result != ERROR_SUCCESS)
+    {
+        std::cerr << "Failed to attach countbytes eBPF program: " << result << std::endl;
+        return 1;
+    }
+
+    five_tuple_t key = {0};
```

Comment:
In C++, the preferred way to default initialize is key{};

In C code you'll need to use = {0};

Reply:
Ohh I see 👍 Changed